### PR TITLE
Make most enumerations into enum classes

### PR DESCRIPTION
**Pull request by binji**

### CODE REVIEW

File: src/tools/wast2wasm.cc [hunk 1]
```diff
@@ -238,5 +238,5 @@ int main(int argc, char** argv) {
 
   wabt_destroy_ast_lexer(lexer);
   wabt_destroy_script(&script);
-  return result;
+  return static_cast<int>(result);
```
Comment (sbc100) [inline src/tools/wast2wasm.cc]: Return result != OK ?  Or are there different result values you want to return from main?

**Comment by binji** [comment]
> Seems like chromium style says it still uses MACRO_STYLE for enum values

True, but it sounds like that was for legacy reasons, the Google C++ style guide says to use `kMixedCase`. Could do that here instead, I suppose.

> Seems like going with that style would result in a much smaller change as none of the test output would change? (maybe?)

Yes, but the names match better with the binary document this way, so... :-) Do you have a preference for `MACRO_STYLE`? I don't like that it looks like it could be a macro. And it's kind of ugly, especially for enum class, where it would be `EnumType::ENUM_VALUE`.

File: src/tools/wast2wasm.cc (hunk 1 above)
Comment (binji) [inline src/tools/wast2wasm.cc]: Done.

**Comment by binji** [comment]
Gonna land this because I think having enum classes is better overall. Let's continue discussion of style on #308.

**Pull request merged**